django/db/backends/base/schema.py

```diff
@@ -594,6 +594,15 @@ def remove_constraint(self, model, constraint):
         if sql:
             self.execute(sql)
 
+    def alter_constraint(self, model, old_constraint, new_constraint):
+        if not self._constraint_should_be_altered(old_constraint, new_constraint):
+            return
+        remove_sql = old_constraint.remove_sql(model, self)
+        add_sql = new_constraint.create_sql(model, self)
+        if remove_sql and add_sql:
+            self.execute(remove_sql)
+            self.execute(add_sql, params=None)
```

Comment:
I thought the current [consensus](https://github.com/django/django/pull/17846#issuecomment-1949371939) was that `AlterConstraint` was always going to be an SQL noop. When SQL needs to be emitted, we would use `RemoveConstraint` and `CreateConstraint` like now.

Did you give some thought to this?

Reply:
Thanks, maybe I misunderstood a little. But can I ask why we are considering AlterConstraint as a no-op, and why we aren't following the example of operations like AlterField, where the operation itself handled changes in the database?

Reply:
We should ask around :-) but I thought it because it implies that it doesn't something distinct from dropping and recreating when in reality it won't.

Reply:
I agree with Jacob. This is not an `ALTER` DDL operation so we should not pretend that it's.